### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by alexandrlonski**
cinema_client

Created several pages, routes and use state manager redux

Ticket: https://trello.com/c/598wuDvL

### CODE REVIEW

File: src/App/App.css [hunk 1]
```diff
@@ -0,0 +1,23 @@
+.button {
+  background-color: #787978; /* Green */
```
Comment (yavorovsky) [inline src/App/App.css:2]: why would you use scss in the project? It's more maintainable and readable

File: src/components/AppRouter/AppRouter.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
+import React from "react";
+import { Container } from "react-bootstrap";
+import { Routes, Route } from "react-router-dom";
+import { authRoutes, publicRoutes } from "../../routes";
+import { isAuth } from "../../data/Data";
+
+const AppRouter = () => {
+  return (
+    <Container style={{ padding: "40px" }}>
```
Comment (yavorovsky) [inline src/components/AppRouter/AppRouter.js:9]: try to keep styles in the same place (.css or here or check some css libraries for react)

File: src/components/AppRouter/AppRouter.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+      <Routes>
+        {isAuth &&
```
Comment (yavorovsky) [inline src/components/AppRouter/AppRouter.js:11]: Really don't like this idea. take a look at High Order Components design. Or check alternatives how to organize routing

File: src/components/Film/Film.js [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+  };
+
+  return (
+    <Card style={{ width: "18rem", margin: "5px" }}>
+      <Card.Img variant="top" src={props.src} />
+      <Card.Body>
+        <Card.Title>{props.title}</Card.Title>
+        <Card.Text>{props.desc}</Card.Text>
+        <NavLink
+          onClick={() => changeFilmId(props.id)}
```
Comment (yavorovsky) [inline src/components/Film/Film.js:22]: First of all, take a look at useCallback. Secondly, you don't need changeFilId func, you just may use () => dispatch(changeFil(filmId)).

File: src/components/Footer/FooterStyles.js [hunk 5]
```diff
@@ -0,0 +1,63 @@
+import styled from "styled-components";
```
Comment (yavorovsky) [inline src/components/Footer/FooterStyles.js:1]: 👍

File: src/components/Header/Header.js [hunk 6]
```diff
@@ -0,0 +1,45 @@
+import React from "react";
+import { Nav, Navbar, Container, Form } from "react-bootstrap";
+import { NavLink } from "react-router-dom";
+import { isAuth } from "../../data/Data";
+import { userHeader, adminHeader, initHeader } from "./HeaderFor";
+
+const Header = () => {
+	return (
+		<Navbar bg="dark" variant="dark">
+			<Container>
```
Comment (yavorovsky) [inline src/components/Header/Header.js:10]: OMG!!!!!. please make the file clear 👎

File: src/components/Header/HeaderFor.js [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+	ADMIN_ROUTE,
+	BASKET_ROUTE,
+	FILMS_ROUTE,
+	LOGIN_ROUTE,
+	REGISTRATION_ROUTE,
+} from "../../utils/consts";
+
+export const initHeader = [
+	{
+		id: 1,
```
Comment (yavorovsky) [inline src/components/Header/HeaderFor.js:12]: the same. I told you not to use "tabs"

File: src/routes.js [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+  {
+    path: ABOUTFILM_ROUTE,
+    Component: AboutFilm,
+  },
+  {
+    path: REGISTRATION_ROUTE,
+    Component: RegistrationPage,
+  },
+  {
+    path: LOGIN_ROUTE,
```
Comment (yavorovsky) [inline src/routes.js:43]: So, if I'm authorized I can log in again?

File: src/routes.js [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+  {
+    path: FILMS_ROUTE,
+    Component: Films,
+  },
+  {
+    path: ABOUTFILM_ROUTE,
+    Component: AboutFilm,
+  },
+  {
+    path: REGISTRATION_ROUTE,
```
Comment (yavorovsky) [inline src/routes.js:39]: or register ?

File: src/store/FilmReducer.js [hunk 10]
```diff
@@ -0,0 +1,19 @@
+const initialSate = {
+  filmId: 1,
+};
+
+const CHANGE_ID = "CHANGE_ID";
```
Comment (yavorovsky) [inline src/store/FilmReducer.js:5]: it would be better move out the file. Imagine you have reducer for 100 handlers. It's gonna be crazy file

Comment (yavorovsky) [reply src/store/FilmReducer.js:5]: Initial state and Constants rather have in separate files but maybe in the same place

**Review by yavorovsky — CHANGES_REQUESTED** [review]

File: src/components/Film/Film.js (hunk 4 above)
Comment (alexandrlonski) [reply src/components/Film/Film.js:22]: Fixed

File: src/components/AppRouter/AppRouter.js (hunk 2 above)
Comment (alexandrlonski) [reply src/components/AppRouter/AppRouter.js:9]: fixed

File: src/App/App.css (hunk 1 above)
Comment (alexandrlonski) [reply src/App/App.css:2]: fixed

File: src/components/Header/Header.js (hunk 6 above)
Comment (alexandrlonski) [reply src/components/Header/Header.js:10]: fixed

File: src/components/Header/HeaderFor.js (hunk 7 above)
Comment (alexandrlonski) [reply src/components/Header/HeaderFor.js:12]: fixed

**Pull request merged**